### ISSUE TITLE
Linter plugins: support turning all of the plugin's rules (or assign a category to the plugin)

### ISSUE THREAD

**Issue by Lippiece** [oxc-project/oxc#14627]
Thank you for your work!

Currently, you can define an eslint plugin and must explicitly define rules you want to have from the plugin.

If you have many plugins and want the majority of their rules, you'll have to define every rule you want, which is tedious. 

Additionally, oxlint supports categories, which allow you to turn on all the rules and disable the ones you don't need. This feature would also be useful for `jsPlugins`.

**Comment by overlookmotel** [comment]
Agreed, the story right now is pretty dreadful! Having to list every rule individually is a pain.

IMO we'd ideally have something like ESLint's presets (e.g. `my-plugin/recommended`). But question is how to integrate this with Oxlint's `categories` for built-in rules. We need to do some thinking about a good design.

**Comment by silvenon** [comment]
For JS plugins I would like something like `"jsExtends": ["eslint-plugin-testing-library/recommended"]` to work. This would also set `jsPlugins: ["eslint-plugin-testing-library"]` (if the `recommended` config sets it).

Alternatively something like:

```json
{
  "jsPlugins": [
    {
      "specifier": "eslint-plugin-testing-library",
      "config": "recommended"
    }
  ]
}
```

that way you would avoid the additional `jsExtends` field.

**Comment by AnderssonPeter** [comment]
Are there any news on this feature?
Going from `eslint` to `oxlint` this causes some frustration when upgrading packages that add/remove rules.
Now you need to keep track of all rules that are added/removed to recommended list and manage it manually, before we just had to upgrade a package and everything just worked out of the box.

**Comment by rose-bouchayer** [comment]
My temporary workaround is a small Node.js script that load the config from the plugin and writes it to a JSON file.

```js
import { writeFile } from 'node:fs/promises'
import stylistic from '@stylistic/eslint-plugin'

async function main() {
  const config = {
    jsPlugins: [
      {
        name: "@stylistic",
        specifier: "@stylistic/eslint-plugin"
      }
    ],
    rules: stylistic.configs.recommended
  };
  const json = JSON.stringify(config, null, 2);
  await writeFile('oxlint.stylistic.json', json, 'utf-8')
}

main();
```

Then it can be extended in `.oxlintrc.json`.

```json
{
  "extends": ["oxlint.stylistic.json"]
}
```

You could have this generated file not versioned and your personal `.oxlintrc.json` configuration overriding some rules.

**Comment by camc314** [comment]
Hi, you can write code such as:


```ts
// oxlint.config.ts
import { defineConfig } from 'oxlint';

export default defineConfig({
  jsPlugins: [
    // setup the plugin here
  ],
  rules: {
    ...someLinterPlugin.configs.recommended.rules,
  },
});
```

for @rose-bouchayer 's use case, you can write code like:

```ts 
// oxlint.config.ts
import { defineConfig } from 'oxlint';

import stylistic from '@stylistic/eslint-plugin'

export default defineConfig({
  jsPlugins: [
    {
      name: "@stylistic",
      specifier: "@stylistic/eslint-plugin"
    }
  ],
  rules: {
    ...stylistic.configs.recommended
  },
});
```

Please note, I didn't test either of these configurations (but both should work) - you might have to move some bits and pieces around slightly to get it working (the types should give you hints in the right direction here.

**Comment by camc314** [comment]
Based on the above, I believe this issue is complete?

I've also opened https://github.com/oxc-project/website/issues/1013 in relation to improving the docs surrouding this

**Comment by merodiro** [comment]
@camc314 What about if I want to enable the recommended config for built in plugins. for example unicorn's recommended config or typescript-eslint's strict or stylistic config?

**Comment by rose-bouchayer** [comment]
Well @camc314 your solution works only for ts config definition. In my case I'm using JSON which is why I needed to generate the file through a Node.js script. So not sure the issue is really solved to me 😕

**Comment by silvenon** [comment]
I also wish I didn't have to go TS for this, that feels more like a workaround than a solution. But I also get that a better solution is very low prio.
